ktor-server/ktor-server-host-common/jvm/src/io/ktor/server/engine/BaseApplicationEngine.kt

```diff
@@ -33,11 +35,20 @@ public abstract class BaseApplicationEngine(
             it.sendPipeline.merge(pipeline.sendPipeline)
             it.receivePipeline.installDefaultTransformations()
             it.sendPipeline.installDefaultTransformations()
+            it.installDefaultInterceptors()
```

Comment:
What about the engines? Don't they setup the same and how does it interfer with this?

Reply:
Engines used to set up it by using `defaultEnginePipeline()`, but the test engine does not use it. I removed 404 handling in real engines here https://github.com/ktorio/ktor/pull/2115/files#diff-304df4306f4d507c42cc92b972a3d7415cc80ae943455cea7411d65dbf2ee0ffL37